tools/accuracy_checker/accuracy_checker/annotation_converters/fewrel.py

```diff
@@ -0,0 +1,389 @@
+                optional=True, default=128, value_type=int
+            ),
+            'max_query_length': NumberField(
+                description='The maximum number of tokens for the question.',
+                optional=True, default=64, value_type=int
+            ),
+            'doc_stride': NumberField(
+                description="When splitting up a long document into chunks, how much stride to take between chunks.",
+                optional=True, default=128, value_type=int
+            ),
+            'lower_case': BoolField(optional=True, default=False, description='Switch tokens to lower case register')
+        })
+
+        return configuration_parameters
+
+    def configure(self):
+        self.testing_file = self.get_value_from_config('testing_file')
+        self.kgentity_file = self.get_value_from_config('kgentity_file')
+        self.max_seq_length = int(self.get_value_from_config('max_seq_length'))
+        self.max_query_length = self.get_value_from_config('max_query_length')
+        self.doc_stride = self.get_value_from_config('doc_stride')
+        self.lower_case = self.get_value_from_config('lower_case')
+        self.tokenizer = get_tokenizer(self.config, self.lower_case)
+        self.support_vocab = 'vocab_file' in self.config
+
+    @staticmethod
+    def _load_examples(file):
+        def _is_whitespace(c):
+            if c == " " or c == "\t" or c == "\r" or c == "\n" or ord(c) == 0x202F:
+                return True
+            return False
+
+        examples = []
+        lines = read_json(file)
+
+        for (i, line) in enumerate(lines):
+            guid = "%s-%s" % ('test', i)
+            for x in line['ents']:
+                if x[1] == 1:
+                    x[1] = 0
+                    #print(line['text'][x[1]:x[2]].encode("utf-8"))
+            text_a = (line['text'], line['ents'])
+            label = line['label']
+            examples.append(
+                {
+                    'id': guid,
+                    'text_a': text_a,
+                    'text_b': None,
+                    'label': label,
+                }
+            )
+
+        return examples
+
+    def convert(self, check_content=False, progress_callback=None, progress_interval=100, **kwargs):
+        examples = self._load_examples(self.testing_file)
+        data = np.load(self.kgentity_file)
+        input_ent = data['input_ent']
+        ent_mask = data['ent_mask']
+
+        annotations = []
+        unique_id = 1000000000
+        # DocSpan = namedtuple("DocSpan", ["start", "length"])
+
+        label_list = set([x['label'] for x in examples])
+        label_list = sorted(label_list)
+        label_map = {label: i for i, label in enumerate(label_list)}
+
+        for (example_index, example) in enumerate(examples):
+            # idx = unique_id + example_index
+            ex_text_a = example['text_a'][0]   # text
+            h, t = example['text_a'][1]        # entities
```

Comment:
could you please give more meaningful names for these variable, if it is possible. it is hard to understand for what they used and I have suspicious that it will be difficult to maintain this code in the future...

Reply:
Ok, I'll provide a code with more meaningfull names. This fragment of code is used for annotation of entity names in example text with # # and $ $ symbols.

Reply:
I can't make more informative names for variables in this code because I don't clearly understand relationship between two entities in dataset item. I know that 2 pieces of text corresponding to entity relations can be marked by pair of $ and pair of #. May be make additional comments in README.md like this:
'label': 'P931',
'text': 'Canadian Forces Base Lahr ( IATA : LHA , ICAO : EDTL , former code EDAN ) was a military operated commercial airport located in Lahr , Germany .',
'ents': [['Q597321', 48, 52, 0.5], ['Q7039', 21, 25, 0.5]]
In this case ents[0] points to "EDTL" substring and ents[1] points to "Lahr" substring. Piece of text pointed by ents[0] marked by pair of #. Piece of text pointed by ents[1] marked by pair of  $. 
After markup result will be like following:  
'Canadian Forces Base _$ Lahr $_ ( IATA : LHA , ICAO : # EDTL # , former code EDAN ) was a military operated commercial airport located in Lahr , Germany .'